rotkehlchen/chain/evm/decoding/rainbow/decoder.py

```diff
@@ -0,0 +1,380 @@
+            sender: ChecksumEvmAddress,
+            out_event: 'EvmEvent',
+    ) -> None:
+        """Decodes a swap transaction from token to native token.
+        Via the first and last transfers logs
+        """
+        try:
+            token_address = out_event.asset.resolve_to_evm_token().evm_address
+        except WrongAssetType:
+            token_address = None
+
+        if token_address is None:
+            log.error(
+                f'Failed to decode swap from token to native token '
+                f'for tranction {transaction.tx_hash.hex()} due the or token_address is null.',
+            )
+            return
+
+        sell_amount = swap_amount = None
+        for tx_log in all_logs:
+            if (
+                tx_log.topics[0] == ERC20_OR_ERC721_TRANSFER and
+                tx_log.address == token_address
+            ):
+                # Find the transfer log from the user wallet to router
+                if bytes_to_address(tx_log.topics[2]) == self.router_address:
+                    sell_amount = int.from_bytes(tx_log.data)
+
+                # Find the transfer log from the router to the swap provider
+                elif bytes_to_address(tx_log.topics[1]) == self.router_address:
+                    swap_amount = int.from_bytes(tx_log.data)
+
+                if swap_amount and sell_amount:
+                    break
+
+        if not (sell_amount and swap_amount):
+            log.error(
+                f'Failed to decode swap from token to token '
+                f'for tranction {transaction.tx_hash.hex()} due the sell_amount '
+                'or swap_amount being null.',
+            )
+            return
+
+        fee_event = self._create_fee_event(
+            tx_hash=transaction.tx_hash,
+            timestamp=transaction.timestamp,
+            fee_raw=sell_amount - swap_amount,
+            fee_asset=out_event.asset.resolve_to_evm_token(),
+            sender=sender,
+        )
+        decoded_events.append(fee_event)
+
+        return
+
+    def _process_swap(self,
+            transaction: 'EvmTransaction',
+            decoded_events: list['EvmEvent'],
+            all_logs: list['EvmTxReceiptLog'],
+    ) -> list['EvmEvent']:
+        sender = transaction.from_address
+        in_event, out_event = self._get_in_out_events(decoded_events, sender)
+        if not (in_event and out_event):
+            log.error(
+                f'Failed to post decoding rainbow {transaction.tx_hash.hex()} '
+                f'due to missing in_event or out_event.',
+            )
+            return decoded_events
+
+        if out_event.asset == self.evm_inquirer.native_token and in_event.asset.is_evm_token():
+            # Try to find if have wrap eth into weth log
+            wrap_log = None
+            for tx_log in all_logs:
+                if tx_log.topics[0] == WETH_DEPOSIT_TOPIC:
+                    wrap_log = tx_log
+                    break
+            else:
```

Comment:
mmmm note that in arbitrum (irc it was arbitrum) and other chains the implementation of WETH is not the same and the events aren't emmited or they are different. You might have issues there with this approach. Isn't it easier to check the amounts of (eth sen't by the router to the rainbow wallet vs the the eth sent by the user to the rainbow router). You can do something like checking both ETH and weth if that reduces the complexity. Just asking since I'm not aware of the details

Reply:
Hmm, ok, so I will create 2 functions _find_weth_deposit_event and _find_weth_withdrawn_event, so in the arbitrum we ca override with the correct logic for this chain